packages/blade/src/components/Typography/BaseText/BaseText.tsx

```diff
@@ -2,8 +2,45 @@ import type { ReactElement } from 'react';
 import getIn from '../../../utils/getIn';
 import { useTheme } from '../../BladeProvider';
 import isEmpty from '../../../utils/isEmpty';
+import makeTypographySize from '../../../utils/makeTypographySize';
+import type { ColorContrast } from '../../../tokens/theme/theme';
+import type { Theme } from '../../BladeProvider';
 import StyledBaseText from './StyledBaseText';
-import type { BaseTextProps } from './BaseText.d';
+
+/**
+ * @template TokenType token type generic
+ * @description Tokenises objects to dot notation strings, eg: `surface.text.normal.lowContrast`
+ */
+type DotNotationStringToken<TokenType> = {
+  [K in keyof TokenType]: `${K & string}.${TokenType[K] extends Record<
+    string,
+    ColorContrast | string
+  >
+    ? Extract<keyof TokenType[K], number | string>
+    : DotNotationStringToken<TokenType[K]>}`;
+}[keyof TokenType];
```

Comment:
**Self note:** 
We are using the slow version here, There is actually a bit of lag I can see on intellisence, We might need to move to faster one if the union is large later on.

Reply:
the faster one is this only right? instead of directly doing keyOf on the object itself?

Reply:
No no the `keyof` version is faster, this one was the slower. 
Anyway I think lets keep it as is for now, we can later on change it if we face performance issues. No need to micro optimize rn.